### PR TITLE
[issue_9] add cli

### LINKED ISSUE #9: Implement CLI
We want to provide a CLI to convert any input spdx file to a `.opossum` file.

### PR DESCRIPTION
**Pull request by meretp**
Since we have reached a first "brealthrough" from spdx document to opossum document, I added a cli to use this functionality. 

fixes #9

I am not sure on how to best test this CLI. One option I could think of is to add a step to the GitHub Action which runs the cli with one of the test files as input and compares the generated files against some "expected_opossum.json" file. But I am not sure how to best compare the files and make the GitHub Action fail if the files don't match. One option might be to use this [GitHub Action](https://github.com/GuillaumeFalourd/assert-command-line-output)  or alternatively use some bash logic. Do you have any thoughts on this?

### CODE REVIEW

**Comment by JakobSchubert** [comment]
> Since we have reached a first "brealthrough" from spdx document to opossum document, I added a cli to use this functionality.
> 
> fixes #9
> 
> I am not sure on how to best test this CLI. One option I could think of is to add a step to the GitHub Action which runs the cli with one of the test files as input and compares the generated files against some "expected_opossum.json" file. But I am not sure how to best compare the files and make the GitHub Action fail if the files don't match. One option might be to use this [GitHub Action](https://github.com/GuillaumeFalourd/assert-command-line-output) or alternatively use some bash logic. Do you have any thoughts on this?

Click comes with a CliRunner that can be used in normal pytest tests: https://click.palletsprojects.com/en/8.1.x/testing/#testing-click-applications
I am not sure if we need to assert the whole files, maybe just asserting that a nonempty file is created at the right location is enough. If we assert the whole file (snapshot test) we might need to adapt the test very often on every small change of something.

File: src/opossum_lib/cli.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+from opossum_lib.file_generation import generate_json_file_from_tree, write_dict_to_file
+from opossum_lib.graph_generation import generate_graph_from_spdx
+from opossum_lib.tree_generation import generate_tree_from_graph
+
+
+@click.command()
+@click.option(
+    "--infile",
+    "-i",
+    help="The file containing the document to be converted.",
```
Comment (JakobSchubert) [inline src/opossum_lib/cli.py:26]: You can add `,type=click.Path()` to get some additional features.
I think there is even `click.Path(exists=True)` that validates that the path specified really exists.
I think you can even do `click.Path(exists=True, convert=pathlib.Path)` to get it automatically converted to a Pathlib path.

File: src/opossum_lib/cli.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    "--outfile",
+    "-o",
+    help="The file path to write the generated opossum document to. The generated file "
+    "will be in JSON format, if the specified file path doesn't match this file "
+    'extension ".json" will be appended.',
+)
+def main(infile: str, outfile: str) -> None:
+    """
+    CLI-tool for converting SPDX documents to Opossum documents.
+    To use, run: 'opossum_lib --infile <input file name> --out <output file path>'
```
Comment (JakobSchubert) [inline src/opossum_lib/cli.py:38]: I would have expected the `opossum_lib --infile <input file name> --out <output file path>` to be auto-generated by click (but I might be wrong).
Having this hardcoded feels fragile: I somebody in the future renames any option, he might forget to adapt the hardcoded string.

File: src/opossum_lib/cli.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+import logging
+import sys
+from pathlib import Path
+
+import click
+from spdx.model.document import Document
```
Comment (JakobSchubert) [inline src/opossum_lib/cli.py:13]: Optional idea: You could do
```suggestion
from spdx.model.document import Document as SpdxDocument
```
to clarify the naming as we now have spdx documents as well as opossum documents.

**Review by JakobSchubert — APPROVED** [review]
Looks good, some small comments.

File: src/opossum_lib/cli.py (hunk 1 above)
Comment (meretp) [reply src/opossum_lib/cli.py:26]: Thanks for the pointer. I added `click.Path(exists=True)`, the parse method from `tools-python` exepects a `str` is input, not a `Pathlib.path`.

File: src/opossum_lib/cli.py (hunk 3 above)
Comment (meretp) [reply src/opossum_lib/cli.py]: Done

**Comment by JakobSchubert** [comment]
Thanks for implementing the review comments! Feel free to merge.

File: tests/test_cli.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        spdx2opossum,
+        [
+            options[0],
+            str(Path(__file__).resolve().parent / "data" / "SPDX.spdx"),
+            options[1],
+            str(tmp_path / "output"),
+        ],
+    )
+
+    print(result.output)
```
Comment (JakobSchubert) [inline tests/test_cli.py:28]: Is this print intended?

Comment (meretp) [reply tests/test_cli.py]: No, I will delete this.

**Pull request merged**